rest/admin_api.go

```diff
@@ -260,17 +260,27 @@ func readReplicationParametersFromJSON(jsonData []byte) (params sgreplicate.Repl
 				err = base.HTTPErrorf(http.StatusBadRequest, "/_replicate sync_gateway/bychannel filter; Missing query_params")
 				return
 			}
-			var channels []string
-			err := json.Unmarshal([]byte(in.QueryParams), &channels)
-			if err != nil {
+
+			//The Channels must be passed as an array of strings
+			if chanarray, ok := in.QueryParams.([]interface{}); ok {
```

Comment:
Unfotunately casting does not work for []interface{} to []string, also we need to catch arrays with mixed types, where most might be strings then a caller passes a number or bool or nested object in the array.

Agree about the fallback to JSON Object form, I'll add that.